### ISSUE TITLE
mysterious behavior, can't use macros.  what should I expect?

### ISSUE THREAD

**Issue by micahscopes** [dplduffy/MidiFighterTwister-Bitwig-Script#10]
long story short: It'd be nice if basic usage and working/not working features were documented!

I got a midi fighter twister off of ebay and have a feeling it's not using the default program.  I'm using Linux and have been unable to get the Midi Fighter Utility to recognize the device under Wine...  so I haven't been able to do a factory reset yet.

The script is doing _something_, but I'm not sure what it's supposed to do.  With the current state of the device, I'm unable to interact with macros.  There is also no text feedback to tell me what's going on.  The only things I've been able to interact with so far are sends and pans, but not device macros.  Is that because device macros aren't enabled in the script?  Or do I need to do a factory reset?

I poked around in the code and saw some references to macros... but I have no idea how to use them!  I've just been pressing all kinds of random side and knob buttons.  The lights change and flash, and do all kinds of impressive effects.  But I can't change macros!

**Comment by dplduffy** [comment]
I never finished this script, which is why I never uploaded the documentation. I also sold my midi fighter twister, so unfortunately this is probably where the script will stay unless someone wants to add onto it.

I just put it up here in pdf and docx format. This should help you out. You'll need to setup the encoders in the utility.  Unfortunately I don't have the setup file with me currently, I will try to upload it tomorrow.  Basically this script uses 4 banks of encoders and all 6 side buttons.  Bank1 encoders  = channels 1-16, bank2 encoders = channels 17-32, bank3 = 33-48, bank4 = 49-64.  The side buttons need to be configured to just send midi CC messages.   Left Top, Middle, Bottom = 8, 9, 10. Right Top, Middle, Bottom = 11, 12, 13.

 I remember the macros and device mode being real strange, but I'm pretty sure they were usable in single display format (explained in the user guide).  There should definitely be text feedback, I just checked the code.  The biggest thing I used this for was the Mixer mode and the Melodic step sequencer.

I'll keep this issue open if you have any questions, just put them here. I'll try to answer them to the best of my ability.
